res/controllers/Pioneer-DDJ-SB3-scripts.js

```diff
@@ -0,0 +1,1791 @@
+};
+
+PioneerDDJSB3.deck4Button = function(channel, control, value, status, group) {
+    if (value) {
+        PioneerDDJSB3.deck4Enabled = true;
+        var bpm = engine.getValue(group, "bpm");
+        PioneerDDJSB3.updateBPM(bpm, group);
+        midi.sendShortMsg(0xB3, 0x02, 0x0);
+    }
+};
+
+PioneerDDJSB3.autoLoopButton = function(channel, control, value, status, group) {
+    if (value) {
+        if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "reloop_toggle", true);
+        } else {
+            engine.setValue(group, "beatloop_activate", true);
+        }
+    }
+};
+
+PioneerDDJSB3.reloopButton = function(channel, control, value, status, group) {
+    if (value) {
+        engine.setValue(group, "reloop_toggle", true);
+    }
+};
+
+PioneerDDJSB3.loadButton = function(channel, control, value, status, group) {
+    if (value) {
+        engine.setValue(group, "LoadSelectedTrack", 1);
+    }
+};
+
+///////////////////////////////////////////////////////////////
+//                          HEADPHONE                      //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSB3.headphoneCueMaster = false;
+
+PioneerDDJSB3.masterCueButton = function(channel, control, value) {
+    if (value) {
+        PioneerDDJSB3.headphoneCueMaster = !PioneerDDJSB3.headphoneCueMaster;
+        PioneerDDJSB3.headphoneMasterUpdate();
+    }
+};
+
+PioneerDDJSB3.headphoneCueButton = function(channel, control, value, status, group) {
+    if (value) {
+        script.toggleControl(group, "pfl");
+        PioneerDDJSB3.headphoneMasterUpdate();
+    }
+};
+
+PioneerDDJSB3.headphoneShiftCueButton = function(channel, control, value, status, group) {
+    if (value) {
+        script.toggleControl(PioneerDDJSB3.deckShiftSwitchTable[group], "pfl");
+        PioneerDDJSB3.headphoneMasterUpdate();
+    }
+};
+
+PioneerDDJSB3.headphoneMasterUpdate = function() {
+    var anyDeckCue = false;
+    var masterCue = PioneerDDJSB3.headphoneCueMaster;
+
+    for (var i = 1; i <= 4; i++) {
+        if (engine.getValue("[Channel" + i + "]", "pfl")) {
+            anyDeckCue = true;
+        }
+    }
+
+    if (masterCue) {
+        if (anyDeckCue) {
+            // 50% master 50% cue
+            engine.setValue("[Master]", "headMix", 0);
+        } else {
+            // 100% master
+            // Check if 1 is all master or all cue
+            engine.setValue("[Master]", "headMix", 1);
+        }
+    } else {
+        // 0% master
+        // Check if 1 is all master or all cue
+        engine.setValue("[Master]", "headMix", -1);
+    }
+};
+
+PioneerDDJSB3.headphoneLevel = new components.Pot({
+    unshift: function() {
+        this.group = "[Master]";
+        this.inKey = "headGain";
+        this.disconnect();
+        this.connect();
+    },
+
+    shift: function() {
+        this.group = "[Master]";
+        this.inKey = "headMix";
+        this.disconnect();
+        this.connect();
+    },
```

Comment:
I'm not sure what I should fix here or what the issue is.  I don't understand.

Reply:
I'm proposing to remove this feature entirely because it conflicts with the master pfl feature.

Reply:
I see. Is there a better way to implement this or should this just not be supported on this controller?

Reply:
I don't think it should be supported to be honest, but I'm still up to debate that. My reasoning is primarily that I don't really see a use case for it when there is already the master pfl button. In addition mapping knob controls to shift also results in poor UX because of takeover issues most of the time.